sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp

```diff
@@ -340,27 +340,31 @@ bool OpenCLC_Supports_Extension(
         "trouble parsing query returned from CL_DEVICE_EXTENSIONS_WITH_VERSION "
         "- extension not followed by colon (:)");
   }
-  colon++; // move it forward
 
-  size_t space = ExtensionByVersionLog.find(' ', colon); // could be npos
+  // Note that VersionPtr is an optional parameter in
+  // ext_oneapi_supports_cl_extension().
+  if (VersionPtr) {
```

Comment:
This should be an early return idiom for readability.

Reply:
If VersionPtr is not nullptr, setting the version information is needed before we return.
I think the current code is more readable than redundantly adding "return true".
```
if (!versionPtr)
  return true;
// all the content inside the if-condition
return true;
```

Reply:
That is *NOT* up to the debate, our coding style requires it be an early return, see https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code.

Reply:
Ok. changed.